engine-tests/src/tests/standalone/call_tracer.rs

```diff
@@ -0,0 +1,350 @@
+    let factory_update = {
+        runner.env.block_height += 1;
+        runner.env.predecessor_account_id = "aurora".parse().unwrap();
+        runner.env.signer_account_id = "aurora".parse().unwrap();
+        let storage = &mut runner.storage;
+        let env = &runner.env;
+
+        let mut tx =
+            standalone::StandaloneRunner::template_tx_msg(storage, env, 0, Default::default(), &[]);
+        tx.transaction = sync::types::TransactionKind::FactoryUpdate(xcc_router_bytes);
+        tx
+    };
+    let outcome = sync::execute_transaction_message(&runner.storage, factory_update).unwrap();
+    test_utils::standalone::storage::commit(&mut runner.storage, &outcome);
+    let set_wnear_address = {
+        runner.env.block_height += 1;
+        let storage = &mut runner.storage;
+        let env = &runner.env;
+
+        let mut tx =
+            standalone::StandaloneRunner::template_tx_msg(storage, env, 0, Default::default(), &[]);
+        tx.transaction = sync::types::TransactionKind::FactorySetWNearAddress(wnear_address);
+        tx
+    };
+    let outcome = sync::execute_transaction_message(&runner.storage, set_wnear_address).unwrap();
+    test_utils::standalone::storage::commit(&mut runner.storage, &outcome);
+
+    // User calls XCC precompile
+    let promise = PromiseCreateArgs {
+        target_account_id: "some_account.near".parse().unwrap(),
+        method: "whatever".into(),
+        args: Vec::new(),
+        attached_balance: Yocto::new(1),
+        attached_gas: NearGas::new(100_000_000_000_000),
+    };
+    let xcc_args = CrossContractCallArgs::Delayed(PromiseArgs::Create(promise));
+    let tx = aurora_engine_transactions::legacy::TransactionLegacy {
+        nonce: signer.use_nonce().into(),
+        gas_price: U256::zero(),
+        gas_limit: u64::MAX.into(),
+        to: Some(xcc_address),
+        value: Wei::zero(),
+        data: xcc_args.try_to_vec().unwrap(),
+    };
+    let mut listener = CallTracer::default();
+    let standalone_result = sputnik::traced_call(&mut listener, || {
+        runner
+            .submit_transaction(&signer.secret_key, tx.clone())
+            .unwrap()
+    });
+    assert!(standalone_result.status.is_ok());
+    assert_eq!(listener.call_stack.len(), 1);
+
+    let trace = listener.call_stack.pop().unwrap();
+    assert_eq!(trace.calls.len(), 1);
+    let subcall = trace.calls.first().unwrap();
+    assert_eq!(subcall.call_type, call_tracer::CallType::Call);
+    assert_eq!(subcall.from, xcc_address);
+    assert_eq!(subcall.to.unwrap(), wnear_address);
+    assert_eq!(U256::from_big_endian(&subcall.output), U256::one());
+
+    runner.close();
+}
+
+/// A convenience function for pulling out a sub-call from a trace.
+/// The `path` gives the index to pull out of each `calls` array.
+/// For example `path == []` simply returns the given `root`, while
+/// `path == [2, 0]` will return `root.calls[2].calls[0]`.
+fn subcall_lense<'a, 'b>(
```

Comment:
Is this function used anywhere else rather than in the tests? If not then should be annotated as `#[cfg(test)]`?